app/models/atmosphere/deployment.rb

```diff
@@ -10,10 +10,24 @@
 # Deployments provide a m:n link between appliances and virtual machines.
 module Atmosphere
   class Deployment < ActiveRecord::Base
+    extend Enumerize
     belongs_to :appliance,
       class_name: 'Atmosphere::Appliance'
 
     belongs_to :virtual_machine,
       class_name: 'Atmosphere::VirtualMachine'
+
+    validates :billing_state, presence: true
+
+    enumerize :billing_state, in: ["initial", "prepaid", "expired", "error"], predicates: true
+
+    after_create :initial_billing
```

Comment:
`initial_billing` method is changing fields values and this callback is invoked. According to the [documentation](http://api.rubyonrails.org/classes/ActiveRecord/Callbacks.html) `after_create` callback is invoked after object is saved into DB. Maybe `before_create` callback should be used here?